File: 445/solution.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        q1, q2 = [], []
        head = ListNode(-1)
        while l1:
            q1.append(l1)
            l1 = l1.next
        while l2:
            q2.append(l2)
            l2 = l2.next
        added = 0
        while q1 or q2:
            val = added
            if q1:
                val += q1.pop().val
            if q2:
                val += q2.pop().val
            tmp = ListNode(val % 10, head.next)
            head.next = tmp
            added = val // 10
        if added:
            return ListNode(added, head.next)
        return head.next

File: 445/solution.go
package _445

type ListNode struct {
	Val  int
	Next *ListNode
}

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	s1 := []int{}
	s2 := []int{}
	for l1 != nil {
		s1 = append(s1, l1.Val)
		l1 = l1.Next
	}
	for l2 != nil {
		s2 = append(s2, l2.Val)
		l2 = l2.Next
	}
	added := 0
	head := ListNode{-1, nil}
	for len(s1) != 0 || len(s2) != 0 {
		val := added
		if len(s1) != 0 {
			val += s1[len(s1)-1]
			s1 = s1[:len(s1)-1]
		}
		if len(s2) != 0 {
			val += s2[len(s2)-1]
			s2 = s2[:len(s2)-1]
		}
		tmp := ListNode{val % 10, head.Next}
		head.Next = &tmp
		added = val / 10
	}
	if added != 0 {
		return &ListNode{added, head.Next}
	}
	return head.Next
}
